src/webapp/components/org-unit-selector/OrgUnitSelector.tsx

```diff
@@ -146,72 +128,45 @@ export const OrgUnitSelector: React.FC<OrgUnitSelectorProps> = React.memo(props
-                <DialogContent dividers>
-                    <DisableableTooltip
-                        title={notAvailableText}
-                        disabled={_(illegalOrgUnits).isNotEmpty()}
-                    >
-                        <FormControlLabel
-                            control={
-                                <Switch
-                                    checked={current.onlyUserOU}
-                                    onChange={toggleUserDataSetsOnly}
-                                    name="userDataSetsOnly"
-                                    color="primary"
-                                    disabled={_(illegalOrgUnits).isNotEmpty()}
-                                />
-                            }
-                            label={i18n.t(
-                                "Select current user's organisation units (and children)"
+                <DialogTitle>
+                    <Box display="flex" gridColumnGap={theme.spacing(1)} alignItems="center">
+                        {label}
+                        <Tooltip
+                            title={i18n.t(
+                                "The children of the Organisation Unit selected will also be taken into account."
                             )}
-                        />
-                    </DisableableTooltip>
-                    {/* Prevent flashing the User on switch is toggled */}
-                    <Box
-                        height={ORG_UNIT_SELECTOR_HEIGHT + 84}
-                        className={current.onlyUserOU ? classes.show : classes.hide}
-                        component="span"
-                        role="presentation"
-                    />
-                    <Box className={current.onlyUserOU ? classes.hide : classes.show}>
+                        >
+                            <InfoOutlinedIcon color="action" />
+                        </Tooltip>
+                    </Box>
+                </DialogTitle>
+
+                <NoPaddingContent dividers className={classes.content}>
+                    <Box>
                         <OrgUnitsSelector
                             api={api}
                             withElevation={false}
                             controls={controls}
                             hideMemberCount={false}
                             fullWidth={false}
                             height={ORG_UNIT_SELECTOR_HEIGHT}
-                            onChange={updateSelected}
-                            selected={current.selectedPaths}
+                            onChange={setCurrentPaths}
+                            selected={currentPaths}
                             singleSelection={true}
-                            selectOnClick={true}
                             initiallyExpanded={initiallyExpanded}
                             selectableLevels={[3, 4, 5, 6]}
+                            rootIds={currentUser.organisationUnits.map(getId)}
```

Comment:
If we make these 2 props immutable, all the props will be immutable and the component may be more efficient (if defined as memo)

Reply:
I forgot, sry. Thx for pointing it!